### PR TITLE
Fix multicodec encoding errors for X25519.

### PR DESCRIPTION
**Pull request by msporny**

### CODE REVIEW

File: lib/X25519KeyAgreementKey2020.js [hunk 1]
```diff
@@ -26,11 +32,28 @@ export class X25519KeyAgreementKey2020 extends LDKeyPair {
...
+    const {publicKeyMultibase, privateKeyMultibase} = options;
+
+    if(!publicKeyMultibase) {
+      throw new TypeError('The "publicKeyMultibase" property is required.');
+    }
+
+    if(!publicKeyMultibase || !_isValidKeyHeader(
+      publicKeyMultibase, MULTICODEC_X25519_PUB_HEADER)) {
+      throw new Error(
+        '`publicKeyMultibase` has invalid header bytes: ' +
```
Comment (davidlehn) [inline lib/X25519KeyAgreementKey2020.js]:
```suggestion
        '"publicKeyMultibase" has invalid header bytes: ' +
```

File: lib/X25519KeyAgreementKey2020.js [hunk 2]
```diff
@@ -26,11 +32,28 @@ export class X25519KeyAgreementKey2020 extends LDKeyPair {
...
+        `"${publicKeyMultibase}".`);
     }
-    this.privateKeyMultibase = options.privateKeyMultibase;
+
+    if(privateKeyMultibase && !_isValidKeyHeader(
+      privateKeyMultibase, MULTICODEC_X25519_PRIV_HEADER)) {
+      throw new Error('privateKeyMultibase` has invalid header bytes.');
```
Comment (davidlehn) [inline lib/X25519KeyAgreementKey2020.js]:
```suggestion
      throw new Error('"privateKeyMultibase" has invalid header bytes.');
```

File: lib/X25519KeyAgreementKey2020.js [hunk 3]
```diff
@@ -159,7 +186,7 @@ export class X25519KeyAgreementKey2020 extends LDKeyPair {
       throw new Error('Source public key is required to convert.');
     }
 
-    const edPubkeyBytes = _multibaseDecode({text: publicKeyMultibase});
+    const edPubkeyBytes = _multibaseDecode(publicKeyMultibase);
```
Comment (dlongley) [inline lib/X25519KeyAgreementKey2020.js]: This doesn't check the key header.

File: lib/X25519KeyAgreementKey2020.js [hunk 4]
```diff
@@ -372,69 +356,45 @@ X25519KeyAgreementKey2020.SUITE_CONTEXT =
...
-  if(!(bytes && bytes[0] === 0xec && bytes[1] === 0x01)) {
-    // eslint-disable-next-line max-len
-    throw new Error('Expecting public key to be "x255519-pub" multicodec formatted [0xec, 0x01, <public key bytes>]');
-  }
-  // Remove the multicodec prefix
-  const publicKeyBytes = bytes.slice(2);
-  return publicKeyBytes;
+function _multibaseDecode(text) {
+  const mcValue = base58btc.decode(text.substr(1));
+  return mcValue.slice(2);
```
Comment (dlongley) [inline lib/X25519KeyAgreementKey2020.js]: Did we want to check the key header here? The above check got removed and isn't performed elsewhere.

**Review by dlongley — APPROVED** [review]
Approving but see comments.

File: lib/X25519KeyAgreementKey2020.js (hunk 3 above)
Comment (msporny) [reply lib/X25519KeyAgreementKey2020.js]: Fixed in 5f79c9c90b7aa35f1d4ea5c889506b0b26c9961e.

File: lib/X25519KeyAgreementKey2020.js (hunk 4 above)
Comment (msporny) [reply lib/X25519KeyAgreementKey2020.js]: Fixed in 5f79c9c90b7aa35f1d4ea5c889506b0b26c9961e.

**Pull request merged**